pkg/factory/factory.go

```diff
@@ -53,11 +53,28 @@ func (rf *resourceFactory) GetResourceServer(rp types.ResourcePool) (types.Resou
 		if prefixOverride := rp.GetResourcePrefix(); prefixOverride != "" {
 			prefix = prefixOverride
 		}
-		return resources.NewResourceServer(prefix, rf.endPointSuffix, rf.pluginWatch, rp), nil
+		policy := rp.GetAllocatePolicy()
+		allocator, err := rf.GetAllocator(policy)
+		if err != nil {
+			return nil, err
+		}
+		return resources.NewResourceServer(prefix, rf.endPointSuffix, rf.pluginWatch, rp, allocator), nil
 	}
 	return nil, fmt.Errorf("factory: unable to get resource pool object")
 }
 
+// GetAllocator returns an instance of Allocator using preferredAllocationPolicy
+func (rf *resourceFactory) GetAllocator(policy string) (types.Allocator, error) {
+	switch policy {
+	case "":
+		return nil, nil
+	case "packed":
+		return resources.NewPackedAllocator(), nil
+	default:
+		return nil, fmt.Errorf("GetSelector(): invalid policy %s", policy)
```

Comment:
```suggestion
		return nil, fmt.Errorf("GetAllocator(): invalid policy %s", policy)
```

Should it be like this?